File: pinyougou-cart-interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import java.util.List;

import com.pinyougou.pojogroup.Cart;

/**
 * 购物车服务接口
 * @author Ly
 *
 */
public interface CartService {

	/**
	 * 添加商品到购物车
	 * @param cartList
	 * @param itemId 商品SKU
	 * @param num 数量
	 * @return
	 */
	public List<Cart> addGoodsToCartList(List<Cart> cartList,Long itemId,Integer num);
	
	/**
	 * 从redis中查询购物车
	 * @param username
	 * @return
	 */
	public List<Cart> findCartListFromRedis(String username);
	
	/**
	 * 向redis添加数据
	 * @param username
	 * @param cartList
	 */
	public void saveCartListToRedis(String username,List<Cart> cartList);
	
	/**
	 * 合并购物车
	 * @param cartList1
	 * @param cartList2
	 * @return
	 */
	public List<Cart> mergeCarList(List<Cart> cartList1,List<Cart> cartList2);
}

File: pinyougou-cart-web/src/main/webapp/js/service/cartService.js
app.service("cartService",function($http){
	
	//获取购物车列表
	this.findCartList=function(){
		return $http.get("/cart/findCartList.do");
	}
	
	//增加减商品数量
	this.addGoodsToCartList=function(itemId,num){
		return $http.get("/cart/addGoodsToCartList.do?itemId="+itemId+"&num="+num);
	}
	
	//求合计数量
	this.sum=function(cartList){
		var totalValue={totalNum:0,totalMoney:0.00};
		for(var i=0;i<cartList.length;i++){
			var cart=cartList[i];
			for(var j=0;j<cart.orderItemList.length;j++){
				var orderItem = cart.orderItemList[j];
				totalValue.totalNum+=orderItem.num;
				totalValue.totalMoney+=orderItem.totalFee;
			}
		}
		return totalValue;
	}
});